### DISCUSSION TITLE
Modeling a relationship that's both 1:1 and 1:Many (or a better way to represent this data)

### DISCUSSION THREAD

**Discussion by pip8786** [prisma/prisma#12788] [Q&A] [answered]
I was trying to sort the results of a query based on a field on a related object and ran into an issue. Here was the schema I was dealing with (simplified to only contain relevant portions):

```prisma
model Bundle {
  id           Int             @id @default(autoincrement())
  contents     BundleContent[]
}

model BundleContent {
  id         Int           @id @default(autoincrement())
  updatedAt  DateTime      @updatedAt
  bundleId   Int
  bundle     Bundle        @relation(fields: [bundleId], references: [id])
  @@index([updatedAt])
}
```

What I was hoping to do was to get Bundles ordered by the latest BundleContent's updatedAt date. Our query currently limits the BundleContent to 1, ordered by updatedAt, so we only get the latest BundleContent back, but we can't sort the bundles themselves by that same date since it's a 1:Many relationship. 

I could add another field that represents the latest BundleContent such as this:
```prisma
model Bundle {
  id           Int             @id @default(autoincrement())
  latestContent BundleContent? @relation("Latest") 
  contents     BundleContent[] @relation("Bundle")
}

model BundleContent {
  id         Int           @id @default(autoincrement())
  updatedAt  DateTime      @updatedAt
  bundleId   Int
  bundle     Bundle        @relation("Bundle", fields: [bundleId], references: [id])
  latestBundleId Int? @unique
  latestBundle   Bundle?        @relation("Latest", fields: [latestBundleId], references: [id])
  @@index([updatedAt])
}
```
but that complicates insertions and updates. What's the best way to represent this data so I can still sort by the latest update? I guess another way would be to add an "updatedAt" date on the Bundle that mirrors the one on the latest BundleContent?

**Comment by austincrim** [comment, marked as answer]
Hey @pip8786 👋 ,

I think adding an `updatedAt` field on the `Bundle` model would be the most straightforward way to do this. You could also potentially use a plain SQL query using [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw).

**Reply by pip8786** [reply]
Ok, sounds good. And is there a way to trigger that being marked updated if there's a change to one of its BundleContents? Or do I just have to do that update myself?

**Reply by austincrim** [reply]
Not automatically, but you could write a [Prisma middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) to encapsulate that behavior.

**Reply by pip8786** [reply]
Oh cool, thanks! Another feature of Prisma I hadn't seen yet.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
